pkg/aws/core.go

```diff
@@ -171,6 +176,38 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 	return response, nil
 }
 
+//returns Placement Object required in ec2.RunInstancesInput
+func getPlacementObj(req *driver.CreateMachineRequest) *ec2.Placement {
+
+	tenancy := "default"
+	affinity := "default"
+	var availZone string
+
+	myannotations := req.Machine.Spec.NodeTemplateSpec.ObjectMeta.Annotations
+
+	for key, value := range myannotations {
+
+		if key == awsPlacementAffinity && value != "" {
+			affinity = value
+		} else if key == awsPlacementAvailabilityZone && value != "" {
+			availZone = value
+		} else if key == awsPlacementTenancy && value != "" {
+			tenancy = value
+		}
+	}
```

Comment:
If these three keys (`awsPlacementAffinity`, `awsPlacementAvailabilityZone`, `awsPlacementTenancy`) are the only keys of interest then we do not have to iterate over all the annotations. Considering this is a map type, the emptiness of the value can be check and directly assigned.

Reply:
Nice catch @AxiomSamarth . I think this suggestion makes sense.